api/capi/src/nntrainer.cpp

```diff
@@ -755,10 +787,21 @@ int ml_train_dataset_set_property(ml_train_dataset_h dataset, ...) {
     ML_TRAIN_GET_VALID_DATASET_LOCKED(nndataset, dataset);
     ML_TRAIN_ADOPT_LOCK(nndataset, dataset_lock);
 
-    d = nndataset->dataset;
+    db = nndataset->dataset;
   }
 
-  returnable f = [&]() { return d->setProperty(arg_list); };
+  returnable f = [&]() {
+    int status = ML_ERROR_NONE;
+    for (auto &d : db) {
+      if (d != nullptr) {
```

Comment:
Upon calling  ml_train_set_dataset_set_property, it is calling setProperty for the underlying dataset which is owned by the handle.